sdk/communication/Azure.Communication.Administration/tests/PhoneNumberAdministrationClient/PhoneNumberAdministrationClientLiveTests.cs

```diff
@@ -253,6 +247,64 @@ public async Task CreateReservation(string? locale)
             Assert.AreEqual(areaCode, reservation.AreaCode);
             Assert.IsNull(reservation.ErrorCode);
             Assert.AreEqual(1, reservation.PhoneNumbers?.Count);
+
+            await client.CancelReservationAsync(reservationOperation.Id);
+        }
+
+        [Test]
+        [TestCase(null)]
+        [TestCase("en-US")]
+        public async Task CreateGeographicalReservation(string? locale)
+        {
+            var client = CreateClient();
+            const string countryCode = "US";
+
+            var pageablePhonePlanGroups = client.GetPhonePlanGroupsAsync(countryCode, locale);
+            var phonePlanGroups = await pageablePhonePlanGroups.ToEnumerableAsync().ConfigureAwait(false);
+
+            var phonePlanGroup = phonePlanGroups.First(group => group.PhoneNumberType == PhoneNumberType.Geographic);
+            var pageablePhonePlans = client.GetPhonePlansAsync(countryCode, phonePlanGroup.PhonePlanGroupId, locale);
+            var phonePlan = (await pageablePhonePlans.ToEnumerableAsync()).First();
+
+            var locationOptionsResponse = await client.GetPhonePlanLocationOptionsAsync(countryCode, phonePlanGroup.PhonePlanGroupId, phonePlan.PhonePlanId);
+            var state = locationOptionsResponse.Value.LocationOptions.Options.First();
+
+            var locationOptionsQueries = new List<LocationOptionsQuery>
+            {
+                new LocationOptionsQuery
+                {
+                    LabelId = "state",
+                    OptionsValue = state.Value
+                },
+                new LocationOptionsQuery
+                {
+                    LabelId = "city",
+                    OptionsValue = state.LocationOptions.First().Options.First().Value
+                }
+            };
+
+            var areaCodes = await client.GetAllAreaCodesAsync(phonePlan.LocationType.ToString(), countryCode, phonePlan.PhonePlanId, locationOptionsQueries);
+            var areaCode = areaCodes.Value.PrimaryAreaCodes.First();
+
+            var reservationOptions = new CreateReservationOptions("My reservation", "my description", new[] { phonePlan.PhonePlanId }, areaCode);
+            reservationOptions.Quantity = 1;
+            var reservationOperation = await client.StartReservationAsync(reservationOptions);
+
+            await reservationOperation.WaitForCompletionAsync().ConfigureAwait(false);
```

Comment:
In playback mode, we should use `WaitForCompletionAsync` with a polling interval of 0 everywhere to speed up our playback tests.

See a similar change I made for JS https://github.com/Azure/azure-sdk-for-js/pull/12863

Reply:
Done